### PR TITLE
[MI-2768] Create a connection display flow of home component to view …

### PR DESCRIPTION
**Pull request by ayusht2810**
…the list of connected channels

image
image
image

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: src/containers/HomePage/tablePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        items: ['MS Teams Channel', 'Mattermost Channel', 'Action'],
+    };
+
+    // Services
+    const {isLoading, data, isError, error, isFetching} = useGetConnectedChannelsQuery();
+
+    const handleDisconnectButton = (teamsChannel: string, mmChannel: string) => {
+        // TODO: complete this function
+        // eslint-disable-next-line no-alert
+        alert('clicked disconnect button ' + teamsChannel + ' ' + mmChannel);
```
Comment (Kshitij-Katiyar) [inline src/containers/HomePage/tablePanel.tsx:25]: i think you can put this in same line comment

File: src/styles/pages/_homePage.scss [hunk 2]
```diff
@@ -19,7 +19,34 @@
 
     &__input-button {
         margin-top: 20px;
```
Comment (Kshitij-Katiyar) [inline src/styles/pages/_homePage.scss]: do this 1 line

File: src/styles/pages/_homePage.scss [hunk 3]
```diff
@@ -19,7 +19,34 @@
...
+    &__table-loader {
+        margin-left: 140px;
+    }
+
+    &__table-title {
+        font-weight: 600;
+        font-size: 16px;
+        width: 178px;
+        height: 21px;
+        margin-left: 10px;
```
Comment (Kshitij-Katiyar) [inline src/styles/pages/_homePage.scss]: same

File: src/containers/HomePage/tablePanel.tsx (hunk 1 above)
Comment (ayusht2810) [reply src/containers/HomePage/tablePanel.tsx:25]: I don't think it's necessary as these will be removed. also I don't think we can put them in single line

File: src/styles/pages/_homePage.scss (hunk 2 above)
Comment (ayusht2810) [reply src/styles/pages/_homePage.scss]: using util classes now

File: src/styles/pages/_homePage.scss (hunk 3 above)
Comment (ayusht2810) [reply src/styles/pages/_homePage.scss]: using util class now

**Review by Kshitij-Katiyar — APPROVED** [review]

File: src/containers/HomePage/tablePanel.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    // States
+    const [rowsData, setRowsData] = useState<ConnectedChannelTableData[]>([]);
+
+    // Table headers
+    const header = {
+        items: ['MS Teams Channel', 'Mattermost Channel', 'Action'],
+    };
+
+    // Services
+    const {isLoading, data, isError, error, isFetching} = useGetConnectedChannelsQuery();
```
Comment (avas27JTG) [inline src/containers/HomePage/tablePanel.tsx]: what's the difference between `isLoading` and `isFetching`?

File: src/containers/HomePage/tablePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            setRowsData([{
+                key: 0,
+                items: [
+                    {content: <div className='msteams-home__table-no-content margin-left-300'>{getEmptyTableContent()}</div>},
+                ],
+            }]);
+            return;
+        }
+
+        if (data) {
```
Comment (avas27JTG) [inline src/containers/HomePage/tablePanel.tsx]:
```suggestion
        } else if (data) {
```
avoid using `return` this ways inside `useEffect`

File: src/containers/HomePage/tablePanel.tsx (hunk 4 above)
Comment (ayusht2810) [reply src/containers/HomePage/tablePanel.tsx]: `isLoading` changes on the first API request when there is no data present whereas `isFetching` for subsequent requests irrespective of whether data is present or not

Comment (avas27JTG) [reply src/containers/HomePage/tablePanel.tsx]: can't we keep only `isFetching`?

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**